src/openrct2/object/ObjectJsonHelpers.cpp

```diff
@@ -0,0 +1,381 @@
+            }
+        }
+        return result;
+    }
+
+    static std::string FindLegacyObject(const std::string &name)
+    {
+        const auto env = GetContext()->GetPlatformEnvironment();
+        auto objectsPath = env->GetDirectoryPath(DIRBASE::RCT2, DIRID::OBJECT);
+        auto objectPath = Path::Combine(objectsPath, name);
+#ifndef _WIN32
+        if (!File::Exists(objectPath))
+        {
+            // UNIX based systems need to search for any files with the same name
+            // due to case sensitivity.
+            auto filter = Path::Combine(objectsPath, "*.dat");
+            auto scanner = std::unique_ptr<IFileScanner>(Path::ScanDirectory(filter, false));
+            while (scanner->Next())
+            {
+                auto relativePath = scanner->GetPathRelative();
+                if (String::Equals(relativePath, name, true))
+                {
+                    objectPath = scanner->GetPath();
+                    break;
+                }
+            }
+        }
+#endif
+        return objectPath;
+    }
+
+    static std::vector<rct_g1_element> LoadObjectImages(IReadObjectContext * context, const std::string &name, const std::vector<sint32> &range)
+    {
+        std::vector<rct_g1_element> result;
+        auto objectPath = FindLegacyObject(name);
+        auto obj = ObjectFactory::CreateObjectFromLegacyFile(objectPath.c_str());
+        if (obj != nullptr)
+        {
+            auto &imgTable = static_cast<const Object *>(obj)->GetImageTable();
+            auto numImages = (sint32)imgTable.GetCount();
+            auto images = imgTable.GetImages();
+            size_t placeHoldersAdded = 0;
+            for (auto i : range)
+            {
+                if (i >= 0 && i < numImages)
+                {
+                    auto &objg1 = images[i];
+                    auto length = g1_calculate_data_size(&objg1);
+                    auto g1 = objg1;
+                    g1.offset = (uint8 *)std::malloc(length);
+                    std::memcpy(g1.offset, objg1.offset, length);
+                    result.push_back(g1);
+                }
+                else
+                {
+                    auto g1 = rct_g1_element{};
+                    result.push_back(g1);
+                    placeHoldersAdded++;
+                }
+            }
+            delete obj;
+
+            // Log place holder information
+            if (placeHoldersAdded > 0)
+            {
+                std::string msg = "Adding " + std::to_string(placeHoldersAdded) + " placeholders";
+                context->LogWarning(OBJECT_ERROR_INVALID_PROPERTY, msg.c_str());
+            }
+        }
+        else
+        {
+            std::string msg = "Unable to open '" + objectPath + "'";
+            context->LogWarning(OBJECT_ERROR_INVALID_PROPERTY, msg.c_str());
+            for (size_t i = 0; i < range.size(); i++)
+            {
+                auto g1 = rct_g1_element{};
+                result.push_back(g1);
```

Comment:
`emplace_back`, I suspect there would be more occurrences to fix

Reply:
Actually, as this is default-constructing the objects, you could simply call `resize()` and have it do the right thing in one go.